packages/date-picker/test/validation.common.js

```diff
@@ -347,9 +353,77 @@ describe('validation', () => {
       expect(datePicker.invalid).to.be.true;
 
       // Try valid value.
-      datePicker.value = '2016-01-01';
+      datePicker.value = '2016-02-01';
       expect(datePicker.validate()).to.be.true;
       expect(datePicker.invalid).to.be.false;
     });
   });
+
+  describe('keyboard navigation', () => {
```

Comment:
Getting some test failures:
<img width="673" alt="Screenshot 2023-11-08 at 16 48 11" src="https://github.com/vaadin/web-components/assets/1222264/33b2c69a-b107-439c-bf2c-e954a15f73ab">

Reply:
I get these same failed tests when running from the current `main` branch. I don't think they are related to this PR.

Reply:
Actually, there is one failing test related to my changes. Fixing now.

Reply:
Updated. Tests should be good other than the ones that already fail on `main`.

Reply:
Running in debug mode (`yarn debug --group date-picker`) still reports test failures in this branch:

<img width="664" alt="Screenshot 2023-11-10 at 11 50 45" src="https://github.com/vaadin/web-components/assets/1222264/894d74fe-44fb-4f0c-9af3-84f3330b2044">

> I get these same failed tests when running from the current main branch. I don't think they are related to this PR.

I'm not getting any test failures in `main` branch

Reply:
<img width="995" alt="Screenshot 2023-11-10 at 9 20 20 AM" src="https://github.com/vaadin/web-components/assets/71635930/203e6775-6dbf-4327-a218-ada753e17778">

That is what I get off `main` with a fresh pull.

Reply:
Interesting. This is what I'm getting instead.

<img width="843" alt="Screenshot 2023-11-10 at 16 28 02" src="https://github.com/vaadin/web-components/assets/1222264/29a67638-ece8-4d51-9272-6185fa0180fc">

I wonder if it's some timezone-related bug.

Reply:
I think it must be. Can we log this as a bug unrelated to this PR then and let this proceed?

Reply:
The new tests need to pass on the CI, otherwise the PR can't be merged. I took a look at what's causing them to fail (on our timezone) and it seems to be due to the test accessing a `Date` object passed to an internal function using [`sinon.spy(datePicker._overlayContent, 'focusDate');`](https://github.com/vaadin/web-components/pull/6603/files#diff-148127a8f2a8c18174508169b4769ea4638c66e371ee662dc8d7ac14f3bebb86R369)

I'd suggest moving the "keyboard-navigation" tests to the `keyboard-navigation.common.js` and using the test helpers used by the other tests in that file. The `describe` block needs to be renamed from "keyboard navigation" in that case, maybe to "disabled dates" or such.

I'll try to debug the existing tests in `main` branch to see if they can be fixed.

Reply:
Fixed the timezone-related test issues (the ones seen in your screenshot) in `main` branch here: https://github.com/vaadin/web-components/pull/6779

Now `main` branch should be green in all time zones.

Reply:
To simulate running the tests on different timezones, you can change the `Location` under `Sensors` in Chrome dev tools:

![Screenshot 2023-11-13 at 17 06 48](https://github.com/vaadin/web-components/assets/1222264/b19f98d6-55eb-41d6-a73d-0ced0fc1ad3a)

Reply:
This is great. I'll get the keyboard tests moved and ping you when I'm ready.

Reply:
Thanks for the tip on using `Sensors` in Chrome to simulate the timezone. Here is also how to do it on the command line:
```
TZ=Europe/Berlin yarn test --group date-picker
```

Reply:
> Here is also how to do it on the command line

Thanks. That's a handy tip as well.

Reply:
@tomivirkki can you check my latest update? All the keyboard tests are fixed, but I still get a discrepancy between Polymer and Lit in the `month-calendar.common` tests, specifically in a new test called `should update value on disabled-by-function date tap`. The Lit test incorrectly allows the date to be tapped and I can't trace down why. Maybe the Lit date picker isn't quite working correctly with this new `isDateDisabled` functionality and I've missed something my implementation further under the hood.

Reply:
LitElement differs from PolymerElement in that the component updates asynchronously after property changes. Fixed the test case by adding an await after `isDateDisabled` is set.

Reply:
Ahh, makes sense. Thanks for fixing the test. That wraps up all the remaining changes so I think we're good for a final review.

Reply:
Aligned the new test cases with the existing ones in the same file. The date-picker keyboard navigation tests in general compare Date objects, which is not ideal :/